steps/step4.py

```diff
@@ -0,0 +1,113 @@
+'''
+Step 4: Creation of ERSST Dataset
+
+Create xarray dataset of ERSST data from NOAA.
+Calculate anomaly of sea surface temperature from the dataset.
+'''
+
+# Standard imports
+import urllib.request
+import shutil
+import os
+import sys
+
+# 3rd party imports
+import xarray as xr
+import requests
+
+# Add the parent directory to sys.path
+current_dir = os.path.dirname(__file__)
+parent_dir = os.path.abspath(os.path.join(current_dir, os.pardir))
+sys.path.append(parent_dir)
+
+# Local imports
+from parameters.data import ERSST_url
+from parameters.constants import start_date, end_date, baseline_start_date, baseline_end_date
+
+
+def sst_dataset(url, start, end):
+    """
+    Downloads ERSST data from a given URL, trims it to specified years, and returns it as an xarray dataset.
+
+    Args:
+    url (str): The URL to download the ERSST data file.
+    start (str): The start date for trimming the dataset (e.g., '1880-01-01').
+    end (str): The end date for trimming the dataset (e.g., '2023-12-31').
+
+    Returns:
+    xr.Dataset: An xarray dataset containing the ERSST data for the specified time range.
+    """
+
+    # Set temporary file name
+    local_file = "sst.mnmean.nc"
```

Comment:
can we pop this into an outputs or results directory?

Reply:
Aha - I see that now. It did not remove it from my project directory when I ran it. I ran the program from the top level directory: ```python main/run.py``` so maybe that caused the issue. 

In any case, not worth adding it to a directory if it's being removed.